src/request.cpp

```diff
@@ -26,9 +26,19 @@
 
 namespace mega {
 
+bool Request::includesFetchingNodes() const
+{
+    return mIncludesFetchingNodes;
```

Comment:
I think we could avoid this flag by using a dynamic cast. Something like:
```suggestion
    return cmds.size() == 1 && dynamic_cast<CommandFetchNodes>(cmds.back());
```

Reply:
nice one. updaing accordingly